skimage/filters/thresholding.py

```diff
@@ -335,15 +337,20 @@ def threshold_li(image):
     >>> thresh = threshold_li(image)
     >>> binary = image > thresh
     """
+    assert_nD(image, 2)
     # Requires positive image (because of log(mean))
-    offset = image.min()
-    # Can not use fixed tolerance for float image
-    imrange = image.max() - offset
-    image -= offset
+    offset = img.min()
+    if offset < 0:
```

Comment:
It does need to be offset regardless, or it's a breaking change. Look at the current version. In 0.11.3, the minimum value is found and subtracted from the input. 

If that was buggy behavior I'm open to hearing why, but unless it was it needs to be `if offset != 0:` for backwards compatiblity.